test/integration/dualstack/dualstack_endpoints_test.go

```diff
@@ -0,0 +1,294 @@
+	}{
+		{
+			name:           "Service IPv4 Only",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv4Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+		},
+		{
+			name:           "Service IPv6 Only",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv6Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicySingleStack,
+		},
+		{
+			name:           "Service IPv6 IPv4 Dual Stack",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv6Protocol, v1.IPv4Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicyRequireDualStack,
+		},
+		{
+			name:           "Service IPv4 IPv6 Dual Stack",
+			serviceType:    v1.ServiceTypeClusterIP,
+			ipFamilies:     []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol},
+			ipFamilyPolicy: v1.IPFamilyPolicyRequireDualStack,
+		},
+	}
+
+	for i, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ns := framework.CreateTestingNamespace(fmt.Sprintf("test-endpointslice-dualstack-%d", i), s, t)
+			defer framework.DeleteTestingNamespace(ns, s, t)
+
+			// Create a pod with labels
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: ns.Name,
+					Labels:    labelMap(),
+				},
+				Spec: v1.PodSpec{
+					NodeName: "fakenode",
+					Containers: []v1.Container{
+						{
+							Name:  "fake-name",
+							Image: "fakeimage",
+						},
+					},
+				},
+			}
+
+			createdPod, err := client.CoreV1().Pods(ns.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create pod %s: %v", pod.Name, err)
+			}
+
+			// Set pod IPs
+			podIPbyFamily := map[v1.IPFamily]string{v1.IPv4Protocol: "1.1.1.1", v1.IPv6Protocol: "2001:db2::65"}
+			createdPod.Status = v1.PodStatus{
+				Phase:  v1.PodRunning,
+				PodIPs: []v1.PodIP{{IP: podIPbyFamily[v1.IPv4Protocol]}, {IP: podIPbyFamily[v1.IPv6Protocol]}},
+			}
+			_, err = client.CoreV1().Pods(ns.Name).UpdateStatus(context.TODO(), createdPod, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to update status of pod %s: %v", pod.Name, err)
+			}
+
+			svc := &v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("svc-test-%d", i), // use different services for each test
+					Namespace: ns.Name,
+					Labels:    labelMap(),
+				},
+				Spec: v1.ServiceSpec{
+					Type:           v1.ServiceTypeClusterIP,
+					IPFamilies:     tc.ipFamilies,
+					IPFamilyPolicy: &tc.ipFamilyPolicy,
+					Selector:       labelMap(),
+					Ports: []v1.ServicePort{
+						{
+							Name:       fmt.Sprintf("port-test-%d", i),
+							Port:       443,
+							TargetPort: intstr.IntOrString{IntVal: 443},
+							Protocol:   "TCP",
+						},
+					},
+				},
+			}
+
+			// create a service
+			_, err = client.CoreV1().Services(ns.Name).Create(context.TODO(), svc, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("error creating service: %v", err)
```

Comment:
i think the common way is to uppercase the first word in a test log message, or at least make it consistent across a test (some a lowercase, some uppercase)